test/es-module/test-esm-specifiers-symlink.mjs

```diff
@@ -0,0 +1,38 @@
+import common from '../common';
+import path from 'path';
+import fs from 'fs/promises';
+import tmpdir from '../common/tmpdir';
+import { spawn } from 'child_process';
+import assert from 'assert';
+
+tmpdir.refresh();
+const tmpDir = tmpdir.path;
+
+await fs.mkdir(path.join(tmpDir, 'subfolder/node_modules/package-a'), { recursive: true });
+
+// Create the following file structure:
+// ├── index.mjs
+// ├── subfolder
+// │   ├── index.mjs
+// │   └── node_modules
+// │       └── package-a
+// │           └── index.mjs
+// └── symlink.mjs -> ./subfolder/index.mjs
+const entry = path.join(tmpDir, 'index.mjs');
+const symlink = path.join(tmpDir, 'symlink.mjs');
+const real = path.join(tmpDir, 'subfolder/index.mjs');
+const packageEntry = path.join(tmpDir, 'subfolder/node_modules/package-a/index.mjs');
+await fs.writeFile(entry, 'import "./symlink.mjs";');
+await fs.writeFile(real, 'export { a } from "package-a/index.mjs"');
+await fs.writeFile(packageEntry, 'export const a = 1;');
+try {
+  await fs.symlink(real, symlink);
+} catch (err) {
+  if (err.code !== 'EPERM') throw err;
+  common.skip('insufficient privileges for symlinks');
+}
```

Comment:
Can we move the skip higher? If we have insufficient permissions, no need to waste time on writing `entry` and `packageEntry` :)